spark/src/test/java/com/matthewrathbone/sparktest/SparkJoinsTest.java

```diff
@@ -0,0 +1,76 @@
+package test.java.com.matthewrathbone.sparktest;
+
+import java.io.File;
+
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import main.java.com.matthewrathbone.sparktest.SparkJoins;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import scala.Tuple2;
+
+import com.google.common.base.Optional;
+import com.google.common.io.Files;
+
+public class SparkJoinsTest implements Serializable {
+	  private transient JavaSparkContext sc;
+	  private transient File tempDir;
+
+	  @Before
+	  public void setUp() {
+	    sc = new JavaSparkContext("local", "SparkJoinsTest");
+	    tempDir = Files.createTempDir();
+	    tempDir.deleteOnExit();
+	  }
+
+	  @After
+	  public void tearDown() {
+	    sc.stop();
+	    sc = null;
+	  }
+	  
+	  @Test
+	  public void sortByKey() {
```

Comment:
can we use the same level of abstraction in this version as we are in the Scala version? It should certainly test all the serialization and mapping to types as well as the group by logic.